src/NewButton/button.tsx

```diff
@@ -0,0 +1,215 @@
+      &:focus:not([disabled]) {
+        border-color: ${get('colors.btn.danger.focusBorder')};
+        box-shadow: ${get('shadows.btn.danger.focusShadow')};
+      }
+
+      &:active:not([disabled]) {
+        color: ${get('colors.btn.danger.selectedText')};
+        background-color: ${get('colors.btn.danger.selectedBg')};
+        box-shadow: ${get('shadows.btn.danger.selectedShadow')};
+        border-color: ${get('colors.btn.danger.selectedBorder')};
+      }
+
+      &:disabled {
+        color: ${get('colors.btn.danger.disabledText')};
+        background-color: ${get('colors.btn.danger.disabledBg')};
+        border-color: ${get('colors.btn.danger.disabledBorder')};
+      }
+    `,
+    invisible: css`
+      color: ${get('colors.accent.fg')};
+      background-color: transparent;
+      border: 0;
+      border-radius: ${get('radii.2')};
+      box-shadow: none;
+    
+      &:disabled {
+        color: ${get('colors.primer.fg.disabled')};
+      }
+      &:focus:not([disabled]) {
+        box-shadow: ${get('shadows.btn.focusShadow')};
+      }
+      &:hover:not([disabled]) {
+        background-color: ${get('colors.btn.hoverBg')};
+      }
+      &:active:not([disabled]) {
+        background-color: ${get('colors.btn.selectedBg')};
+      }
+    `
+  }
+  return style[variant]
+}
+
+const getSizes = (size = 'medium', variant: VariantType = 'default', iconOnly: boolean) => {
+  let paddingTop, paddingLeft
+  switch (size) {
+    case 'small':
+      paddingTop = 3
+      paddingLeft = 12
+      break
+    case 'large':
+      paddingTop = 9
+      paddingLeft = 20
+      break
+    case 'medium':
+    default:
+      paddingTop = 5
+      paddingLeft = 16
+  }
+  if (iconOnly) {
+    paddingLeft = paddingTop + 2
+  }
+  if (variant === 'invisible') {
+    paddingTop = paddingTop + 1
+  }
+  return `
+    padding:${paddingTop}px ${paddingLeft}px;
+  `
+}
+const ButtonBase = styled.button<ButtonBaseProps & {iconOnly: boolean}>`
+  ${buttonBaseStyles}
+  ${props => getVariantStyles(props.theme, props.variant)}
+  ${props => getSizes(props.size, props.variant, props.iconOnly)}
+  ${sizes}
+  ${sx}
+  ${fontSize}
+`
+
+const Button = forwardRef<HTMLAnchorElement | HTMLButtonElement, ComponentProps<typeof ButtonBase>>(
+  ({children, ...props}, forwardedRef) => {
+    const {icon: Icon, leadingIcon: LeadingIcon, trailingIcon: TrailingIcon} = props
+    let iconOnly = !!Icon;
+    const iconWrapStyles = {
+      display: 'inline-block',
+    }
+    return (
+      <ButtonBase ref={forwardedRef} {...props} iconOnly={iconOnly}>
+        {LeadingIcon && (
+          <Box as='span' data-component='leadingIcon' sx={iconWrapStyles} aria-hidden={!iconOnly}>
+            <LeadingIcon />
+          </Box>
+        )}
+        {Icon &&  <Box as='span' sx={{display:'inline-block'}} aria-hidden={!iconOnly}>
+            <Icon />
+          </Box>}
+        <span data-component='text' hidden={Icon? true:false}>{children}</span>
```

Comment:
I think `hidden` shouldn't be in the dom if its false, so would `hidden={Icon}` work here?
Also provide an `id` here for the `button`

output something like:
```
<button aria-labelledby="label-id">
<span id="label-id" hidden>label</span>
</button>
```

Reply:
I think `hidden={Icon ? true: undefined}` would achieve it that